File: app/src/main/java/com/stathis/elmepaunivapp/ui/research/recycler/ResearchAdapter.java
package com.stathis.elmepaunivapp.ui.research.recycler;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;
import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.abstraction.DiffItemCallbackClass;
import com.stathis.elmepaunivapp.listeners.UsefulLinkClickListener;
import com.stathis.elmepaunivapp.ui.research.model.ResearchItem;

public class ResearchAdapter extends ListAdapter<ResearchItem, ResearchViewHolder> {

    private UsefulLinkClickListener listener;

    public ResearchAdapter(UsefulLinkClickListener listener) {
        super(new DiffItemCallbackClass<ResearchItem>());
        this.listener = listener;
    }

    @NonNull
    @Override
    public ResearchViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ResearchViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.holder_research_item_row, parent, false),listener);
    }

    @Override
    public void onBindViewHolder(@NonNull ResearchViewHolder holder, int position) {
        holder.present(getItem(position));
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/research/recycler/ResearchAdapter.kt
package com.stathis.elmepaunivapp.ui.research.recycler

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.stathis.elmepaunivapp.abstraction.DiffItemClass
import com.stathis.elmepaunivapp.callbacks.ElmepaClickListener
import com.stathis.elmepaunivapp.abstraction.LocalModel
import com.stathis.elmepaunivapp.databinding.HolderResearchScreenItemBinding

class ResearchAdapter(private val callback : ElmepaClickListener) : ListAdapter<LocalModel, ResearchViewHolder>(DiffItemClass<LocalModel>()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ResearchViewHolder {
        val view = HolderResearchScreenItemBinding.inflate(LayoutInflater.from(parent.context),parent,false)
        return ResearchViewHolder(view,callback)
    }

    override fun onBindViewHolder(holder: ResearchViewHolder, position: Int) {
        holder.bindData(getItem(position))
    }
}